plugins/color-extract/src/App.tsx

```diff
@@ -105,11 +110,10 @@ export function App() {
                 <>
                     <div className="interface">{colorList}</div>
                     <button
-                        disabled={!supportsGradient}
+                        disabled={!supportsGradient || !isAllowedToSetAttributes}
                         style={{ opacity: supportsGradient ? 1 : 0.5 }}
-                        onClick={() => {
-                            setAsGradient(currentSelection, colors)
-                        }}
+                        onClick={() => setAsGradient(currentSelection, colors)}
```

Comment:
Nit: I'd keep it wrapped, it's a good idea to not return when you don't mean to return 😌 (because in the future `setAsGradient` might start returning something other than `void`, and `onClick` might have special behavior for those cases, resulting in a potential bug)